scalameta/semantic/shared/src/main/scala/scala/meta/semantic/Denotation.scala

```diff
@@ -2,9 +2,11 @@ package scala.meta
 package semantic
 
 import org.scalameta.data._
+import scala.meta.inputs._
+import scala.meta.internal.inputs._
 
-@data class Denotation(flags: Long, name: String, info: String) extends HasFlags {
-  def syntax = s"$flagSyntax $name" + (if (info != "") ": " + info else "")
-  def structure = s"""Denotation($flagStructure, "$name", "$info")"""
+@data class Denotation(flags: Long, name: String, info: String, position: Position) extends HasFlags {
+  def syntax = s"$flagSyntax $name ${position.syntax}" + (if (info != "") ": " + info else "")
```

Comment:
I saw some printouts of `.syntax` in https://travis-ci.org/scalameta/scalameta/jobs/257894597, and I'd like to propose changes to this method:
  1. How about we don't print `position` at all if it's `None`?
  1. How about we add `position` to the very end of `syntax`, not in the middle?